tests/test_blocks.py

```diff
@@ -69,6 +71,129 @@ def test_mpl_kwargs(self):
         assert actual._mpl_kwargs == expected
 
 
+class TestLineBlock:
+    def test_2d_inputs(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(x_grid, y_data)
+
+        assert isinstance(line_block, amp.blocks.Line)
+        npt.assert_equal(line_block.x, x_grid)
+        npt.assert_equal(line_block.y, y_data)
+        assert len(line_block) == len(t)
+
+        assert isinstance(line_block.line, mpl.lines.Line2D)
+        xdata, ydata = line_block.line.get_data()
+        npt.assert_equal(xdata, x)
+        npt.assert_equal(ydata, y_data[0, :])
+
+    def test_update(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(x_grid, y_data)
+        line_block._update(frame=1)
+
+        npt.assert_equal(line_block.line.get_xdata(), x)
+        npt.assert_equal(line_block.line.get_ydata(), y_data[1, :])
+
+    def test_constant_x(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(x, y_data)
+
+        npt.assert_equal(line_block.line.get_xdata(), x)
+        npt.assert_equal(line_block.x[-1], x)
+
+    def test_no_x_input(self):
+        x = np.linspace(0, 1, 10)
+        t = np.linspace(0, 1, 5)
+        x_grid, t_grid = np.meshgrid(x, t)
+        y_data = np.sin(2 * np.pi * (x_grid + t_grid))
+
+        line_block = amp.blocks.Line(y_data)
+
+        expected_x = np.arange(10)
+        npt.assert_equal(line_block.line.get_xdata(), expected_x)
+
+    def test_list_input(self):
+        x_data = [np.array([1, 2, 3]), np.array([1, 2, 3])]
+        y_data = [np.array([5, 6, 7]), np.array([4, 2, 9])]
+        line_block = amp.blocks.Line(x_data, y_data)
+        npt.assert_equal(line_block.y, np.array([[5, 6, 7], [4, 2, 9]]))
+        npt.assert_equal(line_block.x, np.array([[1, 2, 3], [1, 2, 3]]))
+
+    @pytest.mark.xfail(reason="Weird assertion behaviour by numpy")
```

Comment:
Ya, I can't fiure out this assertion behaviour of numpy either. I think an image comparison test might be needed.

Reply:
I think a better way might be something like:
```python
def assert_jagged_arrays_equal(x, y):
    for x, y in zip(x, y):
        npt.assert_equal(x, y)
```
This isn't totally general, but if the arrays we're testing don't have a first dimension which contains a set of numpy arrays, then our input is not as expected anyway.